Maple2.Tools/Collision/HoleCircle.cs

```diff
@@ -10,8 +10,8 @@ public HoleCircle(in Vector2 origin, float innerRadius, float outerRadius) : bas
         hole = new Circle(origin, innerRadius);
     }
 
-    public override bool Contains(in Vector2 point) {
-        return base.Contains(point) && !hole.Contains(point);
+    public override bool Contains(in Vector2 point, float epsilon = 1e-5f) {
+        return base.Contains(point, epsilon) && !hole.Contains(point);
```

Comment:
No epsilon given for the second Contains. Could use the same one, but some use cases might require a separate epsilon for the second one.